embassy-rp/src/gpio.rs

```diff
@@ -127,8 +127,8 @@ impl<'d, T: Pin> Output<'d, T> {
 
     /// Is the output pin set as low?
     pub fn is_set_low(&self) -> bool {
-        // todo
-        true
+        // Reading from SIO: GPIO_OUT gives the last value written.
+        unsafe { self.pin.sio_out().value().read() == 0 }
```

Comment:
`VALUE` is a `u32` with each bit being the value of one pin. You should check just the bit of the current pin, not the whole `VALUE`.